### DISCUSSION TITLE
install problem

### DISCUSSION THREAD

**Discussion by czc1557** [BJCaasenbrood/SorotokiCode#17] [Q&A] [answered]
Hi Author, I'm having a little problem installing sorotoki. When sorotoki build, it was found that the compilation failed. I wonder if you know the reason. Thank you very much.My version of matlab is 2019B.
![1](https://github.com/BJCaasenbrood/SorotokiCode/assets/163147265/f5825fdd-f974-4c38-81f9-5b4cad1df558)

**Comment by BJCaasenbrood** [comment]
Can you navigate to `MATLAB/SorotokiCode/lib/sorotokifem/mex/localsNH` and run `generateMexLNH`? What error message do yo get?

If that doesn't work, try running:

```matlab
codegen LocalsNHFast ...
    -args {coder.typeof(10,[inf,inf]), coder.typeof(10,[inf,inf]), coder.typeof(10,[1,1]), coder.typeof(10,[1,1]), coder.typeof(10,[1,1]), coder.typeof(10,[inf,inf]), coder.typeof(10,[inf,1,inf]), coder.typeof(10,[inf,inf,inf]), coder.typeof(10,[inf,1]), coder.typeof(10,[inf,inf]), coder.typeof(10,[inf,inf]), coder.typeof(10,[1,1]), coder.typeof(10,[1,1]), coder.typeof(10,[inf,1]), coder.typeof(10,[1,1]), coder.typeof(10,[1,1])}
```

**Comment by czc1557** [comment]
Thank you for your reply. My computer is currently not with me, so I will definitely try it when I start work tomorrow.

**Comment by BJCaasenbrood** [comment]
@czc1557 So, I've updated the `CircleCi` config to also include the tests for MATLAB 2019 version. These are the results:

![image](https://github.com/BJCaasenbrood/SorotokiCode/assets/36262141/41ce872f-3156-4199-9689-e4c92d7cee28)

It seems Sorotoki produces an error for `version < 2020b`. I'll look into it. I do highly recommend upgrading MATLAB to 2024A (I know it cumbersome, but your version is 5 years behind).

@czc1557 I would be helpful to show me the output you get from the 2019b version. This might help me with the compatibility issue.

**Comment by czc1557** [comment]
It seems that the version of MATLAB is too low. My classmate's version of MATLAB is 2022A, and he has successfully installed it. I plan to update MATLAB and then try to install SOROTO again. Thank you!

**Reply by BJCaasenbrood** [reply]
Cool, now I also know that there is a version requirement for Sorotoki. Thanks!

**Reply by czc1557** [reply]
Hah，just now, I installed a new version of MATLAB and compiled it successfully. The installation process went very smoothly。Thank you！

**Comment by Beta-y** [comment, marked as answer]
I have similar problems when trying to compile the codes, even after upgrading to MATLAB 2024a. Anyway, I just found that assigning the package path to "MATLAB PATH/toolbox/..." may cause compilation failures. Place them into a separate environment such as "E:\Project" just like @BJCaasenbrood does in the tutorial video could be a good choice.

**Reply by BJCaasenbrood** [reply]
Yeah, I am aware of this. Other users reported the same issue. Having MPI installed via the Add-ons menus causes issues. 

It is just sad that Matlab doesn't have a dedicated package manager like `pip`

**Reply by Beta-y** [reply]
I think that there is another potential reason leading this problem, which is found when I installed this kit on my another pc. In short, there could exist multiple scripts/files named `codegen`, which can be checked using `which codegen -all`. Here are my results:
```
which codegen -all
C:\Softwares\MATLAB_2021b\toolbox\rvctools\robot\demos\codegen.m
C:\Softwares\MATLAB_2021b\toolbox\control\ctrldesign\@tunableSurface\codegen.m  % tunableSurface method
C:\Softwares\MATLAB_2021b\toolbox\coder\matlabcoder\codegen.p                   % Shadowed
```
From above it can be noticed that packages named `rvctools` and `control` do have `codegen.m`. This can be avoided by specifying path of `codegen.p`. For convenience, the following codes work correctly:
```
addpath(pwd); % project path, which cannot be the rootpath of MATLAB toolbox to make sure it can be built, as talked above.
addpath("mpi-packages")
addpath(fullfile(matlabroot,'toolbox/coder/matlabcoder')); % add the path of 'codegen.p' to specify
```
They could be involved into a spearate file named `startup.m`, etc., as conducted by Dr. Caasenbrood did in the tutorial video.
